src/zenml/logging/step_logging.py

```diff
@@ -86,13 +138,99 @@ def emit(self, record: logging.LogRecord) -> None:
             record: The log record to emit.
         """
         try:
-            # Format for storage with timestamp, level, and message
+            # Get level enum
+            level = LoggingLevels.__members__.get(record.levelname.upper())
+
+            # Get the message
             message = self.format(record)
             message = remove_ansi_escape_codes(message).rstrip()
-            self.storage.write(message)
+
+            # Check if message needs to be chunked
+            message_bytes = message.encode("utf-8")
+            if len(message_bytes) <= MAX_MESSAGE_SIZE:
+                # Message is small enough, emit as-is
+                log_record = LogEntry.model_construct(
+                    message=message,
+                    name=record.name,
+                    level=level,
+                    timestamp=utc_now(),
+                    module=record.module,
+                    filename=record.filename,
+                    lineno=record.lineno,
+                )
+                json_line = log_record.model_dump_json(exclude_none=True)
+                self.storage.write(json_line)
+            else:
+                # Message is too large, split into chunks and emit each one
+                chunks = self._split_to_chunks(message)
+                entry_id = uuid4()
+                for i, chunk in enumerate(chunks):
+                    log_record = LogEntry.model_construct(
+                        message=chunk,
+                        name=record.name,
+                        level=level,
+                        module=record.module,
+                        filename=record.filename,
+                        lineno=record.lineno,
+                        timestamp=utc_now(),
+                        chunk_index=i,
+                        total_chunks=len(chunks),
+                        id=entry_id,
+                    )
+
+                    json_line = log_record.model_dump_json(exclude_none=True)
+                    self.storage.write(json_line)
         except Exception:
             pass
 
+    def _split_to_chunks(self, message: str) -> List[str]:
+        """Split a large message into chunks.
+
+        Args:
+            message: The message to split.
+
+        Returns:
+            A list of message chunks.
+        """
+        # Calculate how many chunks we need
+        message_bytes = message.encode("utf-8")
+
+        # Split the message into chunks, handling UTF-8 boundaries
+        chunks = []
+        start = 0
+
+        while start < len(message_bytes):
+            # Calculate the end position for this chunk
+            end = min(start + MAX_MESSAGE_SIZE, len(message_bytes))
+
+            # If we're not at the end of the message, find a safe UTF-8 boundary
+            if end < len(message_bytes):
+                # Back up to find a valid UTF-8 character boundary
+                while end > start + 1 and (message_bytes[end] & 0x80) != 0:
```

Comment:
Is there any difference to doing this compared to the previous logic that tried to decode and caught the exception? To me, that was understandable and made sense, while this implementation is hard to understand and verify that is works